sdk/keyvault/azure-security-keyvault-secrets/src/secret_client.cpp

```diff
@@ -29,240 +29,179 @@ using namespace Azure::Core::Http::Policies;
 using namespace Azure::Core::Http::Policies::_internal;
 using namespace Azure::Security::KeyVault::Secrets::_detail;
 
-std::unique_ptr<RawResponse> SecretClient::SendRequest(
-    Azure::Core::Http::Request& request,
-    Azure::Core::Context const& context) const
-{
-  return KeyVaultSecretsCommonRequest::SendRequest(*m_pipeline, request, context);
-}
-
-Request SecretClient::CreateRequest(
-    HttpMethod method,
-    std::vector<std::string> const& path,
-    Azure::Core::IO::BodyStream* content) const
-{
-  return KeyVaultSecretsCommonRequest::CreateRequest(
-      m_vaultUrl, m_apiVersion, method, path, content);
-}
-
-Request SecretClient::ContinuationTokenRequest(
-    std::vector<std::string> const& path,
-    const Azure::Nullable<std::string>& NextPageToken) const
-{
-  if (NextPageToken)
-  {
-    // Using a continuation token requires to send the request to the continuation token URL instead
-    // of the default URL which is used only for the first page.
-    Azure::Core::Url nextPageUrl(NextPageToken.Value());
-    return Request(HttpMethod::Get, nextPageUrl);
-  }
-  return CreateRequest(HttpMethod::Get, path);
-}
-
 SecretClient::SecretClient(
     std::string const& vaultUrl,
     std::shared_ptr<const Azure::Core::Credentials::TokenCredential> credential,
     SecretClientOptions options)
-    : m_vaultUrl(vaultUrl), m_apiVersion(options.ApiVersion)
 {
-  auto apiVersion = options.ApiVersion;
-  Azure::Core::Url url(vaultUrl);
-
-  std::vector<std::unique_ptr<HttpPolicy>> perRetryPolicies;
-  {
-    Azure::Core::Credentials::TokenRequestContext tokenContext;
-    tokenContext.Scopes = {_internal::UrlScope::GetScopeFromUrl(url)};
-
-    perRetryPolicies.emplace_back(
-        std::make_unique<_internal::KeyVaultChallengeBasedAuthenticationPolicy>(
```

Comment:
Functional regression - we're dropping the `KeyVaultChallengeBasedAuth` policy.
Probably, the easiest path is to inject it into the generated client's code.

Reply:
per chat